source/ranges.tex

```diff
@@ -15835,11 +15835,11 @@
 Initializes \exposid{coroutine_} with
 \tcode{exchange(other.\exposid{coroutine_}, \{\})} and
 \exposid{active_} with
-\tcode{exchange(\brk{}other.active_, nullptr)}.
+\tcode{std::move(\brk{}other.active_)}.
 
 \pnum
 \begin{note}
-Iterators previously obtained from \tcode{other} are not invalidated;
+Iterators previously into \tcode{other} are not invalidated;
```

Comment:
The new phrasing doesn't read like English to me. Also, `other` is an rvalue reference, not a value, so it seems iterators could have been obtained from `other`.

Reply:
I agree that "iterators (previously) into X" is a little vague. Other places (e.g. https://eel.is/c++draft/associative.reqmts.general#115.sentence-2 ) use "iterators pointing to elements of X", but we can't do that here because generators don't have elements. The phrase "iterators into X" is used in https://eel.is/c++draft/sequence.reqmts#37.sentence-3 , as well as on the left-hand side of this diff.

This note could be updated to say "this operation does not modify any iterator `it` in the program, regardless of whether `it.coroutine_ == other.coroutine_` before the operation," but that's _obviously_ tautological, as opposed to being _subtly_ tautological. I think it would also be reasonable to just kill this note entirely.

Re `other` here (in the move-constructor) being an rvalue reference: True, I updated this wording _only_ for consistency with the wording in `operator=`, where `other` _is_ a value (not a reference). I think if the notes are kept, they should be kept parallel, and not with different wording in different places.